docs/tpd_requirements.rst

```diff
@@ -0,0 +1,164 @@
+
+  **Repository Name** - Examples: Snuffy's Cogs, Bobo's Plugins, Cat-Toys, etc
+
+  **Installation Instructions** - Tell the user how to download your repo and access your cogs. Any additional hurdles to using your 
+  creations should be added here as well.
+
+  **Requirements** - Any third party libraries or other requirements you feel necessary for the use of your cogs.
+
+  **Credits** - You must credit anyone whose code you might have used or whom you received a substantial amount of help from.
+
+  **Contact** - How to contact you about bugs, issues, or suggestions. 
+
+  **License** - This is not required, but we do recommend having one.
+
+
+Info Jsons
+==========
+There are two info.json files. One that goes into each cog folder and one at the main level of your repository. Each have similar keys, but serve a different purpose. The following keys are required:
+
+**Repository Level**
+
+ - author
+ - install_msg
+ - name
+ - short
+ - requirements
+ - description
+ - tags
+
+**Cog Level**
+
+ - author
+ - install_msg
+ - name
+ - short
+ - requirements
+ - description
+ - tags
+ - min_python_version
+ - permissions (Senior Only)
+
+You may include as many optional keys as you wish. For more information these keys and their meaning, `read about it here <framework_downloader.html#info-json>`_
+
+.. note::
+
+    If your cog comes bundled with data, or creates any folder or files (outside of config), then this must be disclosed in your 
+    installation message.
+
+Cog Package Folder
+^^^^^^^^^^^^^^^^^^
+The cog folder has the following requirements:
+
+- The folder name, must match the cog.py name. It should be in all lowercase letters
+- The folder should contain an info.json file. Read above for more details on the specific keys needed
+- An __init__.py for the cog
+- The .py file for the cog. (in all lowercase letters)
+- Any helper files or bundled data should be here as well
+
+.. important::
+
+    If your cog comes bundled with data, you are required to put it all into a folder named **data**. You may name any sub-directories 
+    of this folder however you wish. 
+
+********************
+General Requirements
+********************
+These are general requirements that everyone should follow:
+
+- No cog contains malicious code.
+- No stolen or copied code found without proper credits.
+- Disclose if a cog has heavy memory or I/O usage.
+- Disclose if your cog collects any data.
+- Disclose if your cog contains any NSFW material.
+- Disclose if your cog creates or comes bundled with data outside of using Config.
+- Your code is async compatible (not blocking).
+- Does not break the Discord TOS.
+- Does not conflict with core cogs. (E.G., it does not cause a core cog to fail to load).
+- Contains at least three cogs. (See `ctx.send_filtered <tpd_requirements.html#quality-vs-quantity>`__)
+- Does not over-saturate the current list of cogs.
+- Unusable commands are hidden.
+- Default locale must be English.
```

Comment:
I think clarifying the why on this here may be worth it as well, but possibly as an RST referenced footnote given the dual nature of it and fitting into this list.